src/toolkits/object_detection/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -124,6 +124,44 @@ boost::gil::rgba8_image_t::view_t create_starter_image_view(flex_image &object_i
   return starter_image_view;
 }
 
+std::pair<flex_image,flex_dict> create_synthetic_image_from_background_and_starter(
+  flex_image starter,
+  const flex_image &background,
+  std::string label,
+  size_t seed,
+  size_t row_number) {
+  if (!(starter.is_decoded())) {
+    decode_image_inplace(starter);
```

Comment:
This implies that we're still decoding each starter image n times, where n is the number of background images.

Reply:
Should be fixed now.